In [1]:
import pandas as pd

In [3]:
downtown = pd.read_csv('data/dt_data.csv')

In [6]:
region = pd.read_csv('data/region_data.csv')

Table 1
CNS 1 - 20

In [7]:
job_categories = ['Agriculture, Forestry, Fishing and Hunting',
                 'Mining, Quarrying, and Oil and Gas Extraction',
                 'Utilities','Construction','Manufacturing','Wholesale Trade',
                  'Retail Trade','Transportation and Warehousing','Information',
                  'Finance and Insurance','Real Estate and Rental and Leasing',
                  'Professional, Scientific, and Technical Services','Management of Companies and Enterprises',
                  'Administrative and Support and Waste Management and Remediation Services',
                  'Educational Services','Health Care and Social Assistance','Arts, Entertainment, and Recreation',
                  'Accommodation and Food Services','Other Services except Public Administration',
                  'Public Administration']

In [8]:
len(job_categories)

20

Downtown Querying

In [9]:
# Create an empty dictionary to store the sums
dt_sums = {}

# Loop through the column names from "CNS01" to "CNS20"
for column_name in [f'CNS{i:02d}' for i in range(1, 21)]:
    column_sum = downtown[column_name].sum()
    dt_sums[column_name] = column_sum

# Now, column_sums is a dictionary where keys are column names like "CNS01", "CNS02", ..., "CNS20",
# and values are the sums of the respective columns
print(dt_sums)

{'CNS01': 48, 'CNS02': 0, 'CNS03': 2056, 'CNS04': 5460, 'CNS05': 6960, 'CNS06': 2968, 'CNS07': 7060, 'CNS08': 25788, 'CNS09': 8884, 'CNS10': 14972, 'CNS11': 7168, 'CNS12': 57964, 'CNS13': 3412, 'CNS14': 12552, 'CNS15': 7948, 'CNS16': 15880, 'CNS17': 3744, 'CNS18': 41664, 'CNS19': 10388, 'CNS20': 58376}


In [10]:
dt_df = pd.DataFrame.from_dict(dt_sums, orient='index', columns=['Sum'])
dt_df.index = job_categories
dt_df = dt_df.rename(columns={"Sum": "Total jobs"})

In [11]:
total_jobs_dt = dt_df['Total jobs'].sum()
total_jobs_dt

293292

Regional Querying

In [12]:
# Create an empty dictionary to store the sums
regional_sums = {}

# Loop through the column names from "CNS01" to "CNS20"
for column_name in [f'CNS{i:02d}' for i in range(1, 21)]:
    column_sum = region[column_name].sum()
    regional_sums[column_name] = column_sum

# Now, column_sums is a dictionary where keys are column names like "CNS01", "CNS02", ..., "CNS20",
# and values are the sums of the respective columns
print(regional_sums)

{'CNS01': 35824, 'CNS02': 1368, 'CNS03': 30712, 'CNS04': 349096, 'CNS05': 457936, 'CNS06': 178728, 'CNS07': 507012, 'CNS08': 125956, 'CNS09': 97792, 'CNS10': 188896, 'CNS11': 111416, 'CNS12': 624960, 'CNS13': 108064, 'CNS14': 380980, 'CNS15': 510272, 'CNS16': 834016, 'CNS17': 96980, 'CNS18': 492932, 'CNS19': 177640, 'CNS20': 212644}


In [13]:
region_df = pd.DataFrame.from_dict(regional_sums, orient='index', columns=['Sum'])
region_df.index = job_categories
region_df = region_df.rename(columns={"Sum": "Total jobs"})

In [14]:
total_jobs_region = region_df['Total jobs'].sum()
total_jobs_region

5523224

Table 1 Code

In [15]:
dt_df['% of jobs'] = (dt_df['Total jobs'] / total_jobs_dt * 100)
dt_df = dt_df.sort_values(by='% of jobs',ascending = False)
dt_df['% of jobs'] = dt_df['% of jobs'].apply(lambda x: f'{x:.2f}%')

# Who works and lives in the Downtown San Diego employment center?

## Highlight Table 1
Of the 293,292 jobs in
the Downtown San Diego
employment center, the top
five are categorized as Public Administration, Professional, Scientific, and Technical Services,
Accommodation and Food Services	, Transportation and Warehousing, and health care and
Social assistance. Compared
to the region overall, there
is an overrepresentation of
accommodations and
local government and an
underrepresentation of retail.

In [16]:
dt_df.head(10)

Total jobs % of jobs
Public Administration                                    58376    19.90%
Professional, Scientific, and Technical Services         57964    19.76%
Accommodation and Food Services                          41664    14.21%
Transportation and Warehousing                           25788     8.79%
Health Care and Social Assistance                        15880     5.41%
Finance and Insurance                                    14972     5.10%
Administrative and Support and Waste Management...       12552     4.28%
Other Services except Public Administration              10388     3.54%
Information                                               8884     3.03%
Educational Services                                      7948     2.71%

In [17]:
dt_df.to_csv('output/DowntownJobDistribution.csv')

## Figure 1

In [18]:
college_degree_jobs_dt = downtown['CD04'].sum()
college_degree_jobs_region = region['CD04'].sum()

In [19]:
percentage_college_dt = (college_degree_jobs_dt / total_jobs_dt) * 100
percentage_college_region = (college_degree_jobs_region / total_jobs_region) * 100
formatted_percentage_dt = f"{percentage_college_dt:.2f}%"
formatted_percentage_region = f"{percentage_college_region:.2f}%"

In [20]:
'Downtown College Degree Percent: ' + formatted_percentage_dt

'Downtown College Degree Percent: 26.59%'

In [21]:
'Region College Degree Percent: ' + formatted_percentage_region

'Region College Degree Percent: 24.53%'

## Around 27% of workers in the downtown employment center have a 4-year college degree (or higher), a little higher than the regional average (24%).